src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const result = [];
+
+  for (const item of actions) {
+    switch (item.type) {
+      case 'addProperties':
+        newState = { ...newState, ...item.extraData };
+        result.push(newState);
+        break;
+
+      case 'removeProperties':
+        newState = { ...newState };
```

Comment:
There is no need to create a new copy of 'newState' here since it's already a separate object from the original 'state'.